Rebase

benoitf opened the pull request:
rebase from upstream changes

lszomoru committed:
SCMActionButton overflow improvements

hediet committed:
Renaming & docs update.

lszomoru committed:
Replace path.relative with a helper function that factors in case sensitivity but preserve the casing of the resource path

hediet committed:
Abstracts ITextModel to ITokenizerSource.

connor4312 committed:
smoketests: use new extractSync option

sbatten committed:
add notification

sbatten committed:
tweak wording

sandy081 committed:
#91286 use throttled delayer for auto sync

sbatten committed:
fixes #141097

Tyriar committed:
Move capabilities back to common, fixing layering problems

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

roblourens committed:
Don't wait for missing search providers when some search provider is present
Fix #139172

Tyriar committed:
Fix build

Tyriar committed:
Fix windows pwsh shell integration arg injection

Fixes #141098

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

Tyriar committed:
Fix tests after refactor

Tyriar committed:
Fix cwd ordering

Fixes #141008

meganrogge committed:
warn for custom shell args that shell integration won't be enabled (#141028)

TylerLeonhardt committed:
refactor credentialsmainservice and ensure the right one is used (#141100)

lramos15 committed:
Reintroduce telemetry proposal

sandy081 committed:
#91286 handle too many requests on the client

meganrogge committed:
correct the link scanning order

sbatten committed:
add customize layout quick pick

rebornix committed:
no view ref in notebookBrowser

rebornix committed:
IOutputRenderer.

rebornix committed:
simpler typings in NotebooKernelActionViewItem

rebornix committed:
partially revert 8000dc92.

rebornix committed:
:lipstick:

sbatten committed:
upper l

sbatten committed:
hide menu option when using mac native

sbatten committed:
add layout modes

connor4312 committed:
testing: only reveal testing view for ui-triggered runs

roblourens committed:
Fix uncaught exception during unit test. Fix #141018

mjbvz committed:
Make sure we convert webview view columns if needed

Fixes #138914

Commit:
Delete build-chat.yml

Extra slack noise from community PR's not helpful, we get notifications for Azure CI failures anyways.

bpasero committed:
smoke - skip flaky one (#141054)

TylerLeonhardt committed:
When keytar fails to be used, use an in-memory credential store (#141120)

* have inmemory fallback

* move InMemoryCredentialsProvider to common for shared scenarios

KendallDoesCoding committed:
creating pr's section

bpasero committed:
smoke - properly skip tests (#141054)

chrmarti committed:
Update name description (vscode-remote-release#6143)

bpasero committed:
api - clarify path casing for file watchers

bpasero committed:
web - introduce `web.api.ts`

aeschli committed:
no remote indicator for code server web

sandy081 committed:
#91286 :lisptick:

bpasero committed:
web - remove `workbench.web.main`

bpasero committed:
web - API :lipstick:

bpasero committed:
web - update layer rules

alexdima committed:
Add explicit layers for editor contributions (fixes #140995)

jrieken committed:
document that async-dipose functions aren't awaited, https://github.com/microsoft/vscode/issues/140697

sandy081 committed:
Merge branch 'main' into sandy081/syncTasks

bpasero committed:
web - move some interfaces to their sources

isidorn committed:
update distro

aeschli committed:
adopt --server-data-dir

hediet committed:
Implements #136452 by introducing the flag completeBracketPairs for inline completions.

bpasero committed:
web - separate factory from API

alexr00 committed:
Comment detail should be timestamp
Part of #139524

alexdima committed:
Refuse launching unless a connection token is provided

bpasero committed:
web - cleanup factory

lszomoru committed:
Revert "Replace path.relative with a helper function that factors in case sensitivity"

This reverts commit c3bd29b3d09e29b49f8050a5b55106960a671e3b.

bpasero committed:
web - lift some `browser` specific things to `browser` layer

bpasero committed:
debt - introduce browser environment service with access to workbench options

bpasero committed:
Merge branch 'main' into ben/web-api-layers

alexr00 committed:
Fix string comment body displayed as markdown
Fixes #140967

aeschli committed:
Merge branch 'main' into aeschli/noremoteindicatorweb

bpasero committed:
web - clean up web environment service

aeschli committed:
move remote and web tests to scripts folder

hediet committed:
Updates sound files by louder ones.

hediet committed:
Play folded area sound only for folded areas.

Commit:
fix typo

alexdima committed:
Merge pull request #141147 from microsoft/alex/mandatory-connection-token

Refuse launching unless a connection token is provided

alexr00 committed:
Offer to create tasks.json when only user tasks
Fixes #140867

aeschli committed:
only call yarn gulp node when needed

alexdima committed:
Move contribution interface to `/extensions/` (#141154)

Tyriar committed:
Use correct scroll settings on terminal init

Fixes #140788

meganrogge committed:
add test for link with relative cwd (#141101)

meganrogge committed:
test terminal getLinks and open recent commands (#141119)

aeschli committed:
show localhost again in Web UI available message

alexdima committed:
Move `ExtensionKind` to `platform/environment` (#141154)

lramos15 committed:
Add initial telemetry server arg

alexdima committed:
Move virtual workspace functions to `platform/workspace` (#141154)

alexdima committed:
Complete implementation of `--without-connection-token` (closes #136615)

jrieken committed:
use adaptive `IFeatureDebounceInformation` for folding, https://github.com/microsoft/vscode/issues/140557

Tyriar committed:
Test sequences get passed along to capabilities

aeschli committed:
add missing semicolon

Tyriar committed:
Tidy up shell integration addon

alexdima committed:
Move tunnel service to its own folder (#141154)

lramos15 committed:
initial-telemetry-level -> telemetry-level

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

bpasero committed:
Merge branch 'main' into ben/web-api-layers

bpasero committed:
fix paths

aeschli committed:
Merge branch 'main' into aeschli/noremoteindicatorweb

bpasero committed:
Merge pull request #141135 from microsoft/ben/web-api-layers

Layer breaker in `environmentService.ts`

aeschli committed:
server-cli: typo in compatibility message

aeschli committed:
Merge branch 'main' into aeschli/noremoteindicatorweb

aeschli committed:
always set connection-token

bpasero committed:
debt - move `watermar.css` into `media` folder

aeschli committed:
test resolver: prefer connection-token over connection-token-file

jrieken committed:
:lipstick:

Tyriar committed:
Fix a relative import warning

sbatten committed:
move css file to media folder

roblourens committed:
Update milestone on endgame notebooks

sbatten committed:
tweak panel alignment command text based on feedback

sandy081 committed:
rename includePreRelease flag to preRelease for extensionInfo

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

meganrogge committed:
use actual backend OS when resolving shell integration args

Tyriar committed:
Merge command trackers into single addon using capabilities

Tyriar committed:
Active partial command detection capability

ChaseKnowlden committed:
Bump @vscode/windows-registry to 1.0.5 in remote (#140599)

* :up: @vscode/windows-registry@1.0.5

* minor fixes

* Alias vscode-windows-registry

Tyriar committed:
Debug log when capabilities change

TylerLeonhardt committed:
check type. fixes https://github.com/microsoft/vscode-internalbacklog/issues/2581

sbatten committed:
upade distro

Tyriar committed:
Register partial command detection capability in tracker test

Tyriar committed:
Move TerminalCapability into workbench

Tyriar committed:
Fix layer issue

Tyriar committed:
Fix compile

Tyriar committed:
Pass word link tests

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

Tyriar committed:
Remove unused event

Tyriar committed:
Fix xtermterminal test

alexdima committed:
Move `editorColorRegistry` to `/core/` (#141174)

sandy081 committed:
Fix #139015
- make single query to get all compatible extensions while checking updates

sandy081 committed:
fix #138128

sbatten committed:
fixes #138350

alexdima committed:
Move change interfaces to `diffComputer` (#141174)

sbatten committed:
fixes #138653

sbatten committed:
remove unused import

alexdima committed:
Move `markdownRenderer` to `/contrib/` (#141174)

alexdima committed:
Extract `StableEditorScrollState` to a separate file (#141174)

alexdima committed:
Move `editorState` to `/contrib/` (#141174)

alexdima committed:
Remove `editor/browser/core` (#141174)

alexdima committed:
Move `ISingleEditOperation` to `/core/` (#141174)

Tyriar committed:
Get cwd word link tests passing

alexdima committed:
Move `token.ts` to `/languages/` (#141174)

connor4312 committed:
testing: show total to be run when running tests

alexdima committed:
Rename `editor/common/controller` to `editor/common/cursor` (#141174)

sbatten committed:
fixes #140048

rebornix committed:
move folding model into core.

alexdima committed:
Remove `/view/` dependency in `textModel.ts` (#141174)

alexdima committed:
Move `prefixSumComputer.ts` to `/model/` (#141174)

sandy081 committed:
fix tests

alexdima committed:
Move `wordCharacterClassifier` to `/core/` (#141174)

alexdima committed:
Move `cursorColumns` to `/core/` (#141174)

sandy081 committed:
Merge branch 'sandy081/syncTasks'

alexdima committed:
Move `EDITOR_MODEL_DEFAULTS` to `textModel.ts`

Tyriar committed:
Merge remote-tracking branch 'origin/main' into tyriar/shell_tests

alexdima committed:
Move editor base api to `/services/` (#141174)

alexdima committed:
Adjust exemption path

meganrogge committed:
with -> when

rebornix committed:
tangle: notebook folding.

Tyriar committed:
Correct get output by using cmd start and finish markers

connor4312 committed:
testing: remember tree state better, don't expand by default

Fixes #140332

Tyriar committed:
Merge pull request #141083 from microsoft/tyriar/shell_tests

Shell integration tests and refactors

Tyriar committed:
Remove unneeded TODO

Tyriar committed:
Ensure args check/warn only for pwsh on Windows

alexdima committed:
Move `viewEvents` and `viewContext` to `/viewModel/` (#141174)

lramos15 committed:
Remove unused keyboard layout fragment

alexdima committed:
Move `IDiffComputationResult` to `/diff/` (#141174)

Tyriar committed:
Partial command detect unit tests, fix event

alexdima committed:
Move `IEditorConstructionOptions` to `/config/` (#141174)

rebornix committed:
extract notebook contextkeys

alexdima committed:
Move `wordHelper.ts` and extract `EDITOR_MODEL_DEFAULTS` to `/core/` (#141174)

alexdima committed:
Move `overviewZoneManager` to `/viewModel/` (#141174)

Tyriar committed:
Add start of cmd detection tests

Part of #140668

Tyriar committed:
Remove some timestamp code from test

alexdima committed:
Move `renderingContext` to `/browser/`

rebornix committed:
CellReveal into notebookCellList

alexdima committed:
Extract `IDimension` to `/core/` (#141174)

connor4312 committed:
testing: fix slowness when changes state for large numbers of tests

Fixes #140606
Fixes #140745

alexdima committed:
Move `/model/tokens/` to `/tokens/` (#141174)

connor4312 committed:
testing: additional fixes for last commit

rebornix committed:
split notebookBrower test.

mjbvz committed:
Pick up new minor TS version

Tyriar committed:
Avoid array ref check when injecting shell args

Fixes #141189

alexdima committed:
Reduce `/model/` deps in `model.ts` (#141174)

alexdima committed:
Fix recipe file

alexdima committed:
Use `CursorColumns` directly

TylerLeonhardt committed:
finalize quickpick API

Tyriar committed:
Improve bash shell integration script

- Adds a welcome message (trying out)
- Improve mechanism for wrapping function
- Run bashrc manually as it doesn't happen with --init-file

Fixes #140933

alexdima committed:
Remove dependencies to `/languages/` in `languages.ts` (#141174)

rebornix committed:
notebook view events.

Tyriar committed:
First test for cmd detection

Part of #140668

alexdima committed:
Move `languageSelector` out of `/languages/` (#141174)

sbatten committed:
fixes #140648

Tyriar committed:
Fix capabilities not showing up sometimes

Part of #141160

rebornix committed:
NOTEBOOK_EDITOR_ID.

Tyriar committed:
Listen to instance capability change in tabs

Fixes #141160

rebornix committed:
tangle notebook rendering common.

TylerLeonhardt committed:
add doc comments for new quick pick separator API

meganrogge committed:
xtermTerminal -> linkManager (#141172)

meganrogge committed:
improve shellIntegration handling of arguments, add tests (#141197)

bpasero committed:
Hard to compare two files of the same name (fix #141132)

bpasero committed:
Smoke test runner fails or hangs without error (fix #141148)

bpasero committed:
welcome - introduce dedicated top level folders (#141138)

* welcome - introduce dedicated top level folders

* Remove special rules for `contrib/welcome`

* fix svgs

* fix layers lint

Co-authored-by: Alex Dima <alexdima@microsoft.com>

bpasero committed:
User-data file watching is broken in shared process (fixes #141176)

bpasero committed:
Cleanup package.json test scripts (#141210)

* tests - remove `yarn test`

* fix smoke test

* readme

* :lipstick:

* ci - do not install playwright

sandy081 committed:
fix sync tests

sandy081 committed:
Fix #140949

sandy081 committed:
#139015 more optimizations

sandy081 committed:
#139015 more refactorings
- use getExtensions for querying by id or name
- introduce getExtensions in IExtensionsWorkbenchService

bpasero committed:
debt - cleanup context key handling in workbench

sandy081 committed:
fix #139134

sandy081 committed:
Fix #134246

sandy081 committed:
fix #140029

sandy081 committed:
Fix #137770

sandy081 committed:
Fix #122457

sandy081 committed:
Fix #138527

sandy081 committed:
fixes
- do not wait for recommendation prompt
- close recommendations view if empty

sandy081 committed:
Fix #138673

sandy081 committed:
Fix #140794

sandy081 committed:
Fix #125089

bpasero committed:
tests - fix bad `vscode` import

bpasero committed:
editors - understand untitled with associated resources for `--wait` (#141237)

jrieken committed:
use correct cancellation token type, fyi @sandy081

hediet committed:
Lifts getMetadata & getLanguageId to IViewLineTokens.

gjsjohnmurray committed:
Support --#region and --#endregion in SQL (#141257)

jrieken committed:
fix https://github.com/microsoft/vscode/issues/140138

joaomoreno committed:
bump min tree indent to 4

aeschli committed:
test-web-integration: install test/integration/browser & playwright

jrieken committed:
fix https://github.com/microsoft/vscode/issues/140907

aeschli committed:
reenable 'vscode API - languages - link detector' Fixes #140878

aeschli committed:
Merge pull request #141130 from KendallDoesCoding/patch-2

creating pr's section

aeschli committed:
Merge pull request #141162 from microsoft/joh/adaptiveFolding

use adaptive `IFeatureDebounceInformation` for folding

joaomoreno committed:
missing terrapin usage

joaomoreno committed:
fixes #141067

jrieken committed:
fix https://github.com/microsoft/vscode/issues/139410

sandy081 committed:
remove unused telemetry events

bpasero committed:
Rename `workbench.web.api.*` to `workbench.web.main.*` (#141263)

* web - rename entry file

* web - rename entry file also for server

* rename main entry in copy

aeschli committed:
Merge pull request #141264 from gjsjohnmurray/fix-141257

Support `--#region` and `--#endregion` in SQL (#141257)

alexdima committed:
Do not force that a connection token CLI argument is present

alexdima committed:
Rename `remoteExtensionHostAgent` to `server.main`

alexdima committed:
Rename `remoteCli` to `server.cli`

alexdima committed:
Reuse `extensionHostProcess` entry point in the server case

alexdima committed:
Remove `vs/workbench/workbech.web.api` (no need for backwards compat in `reh-web`)

alexdima committed:
Remove `server.sh` (no need for backwards compat in `reh-web`)

aeschli committed:
wsl code: pass SERVERDATAFOLDER

sandy081 committed:
show message if an extension has no other versions

hediet committed:
Fixes compile errors.

Timmmm committed:
Fix parseAddress regex (#140977)

The previous regex was not very good and meant that hostnames containing digits and full stops were not accepted. This one should be more accurate. You can test it [here](https://regex101.com/r/ZVItcq/1).

Because regex's are completely unreadable here's how it breaks down:

```
^    - Start
(     - Hostname group
  [a-zA-Z0-9-]+    - First domain component
  (?:                         - Optional subsequent components
     \.[a-zA-Z0-9-]+     - Each subsequent component is a . followed by the component
  )*                      - Any number of subsequent components
:     - Port separator
)?   - Hostname group is optional
([0-9]+)    - Port, which is mandator
$   - End
```

lszomoru committed:
Tweak sync button style

joaomoreno committed:
fix relative import

joaomoreno committed:
fix imports

hediet committed:
Merge pull request #141146 from microsoft/hediet/inline-completions-bracket-completion

inline completions bracket completion

sandy081 committed:
Fix #141096

jrieken committed:
allow to pass max-delay to debounce info, defaults to min^2 millis, https://github.com/microsoft/vscode/issues/140557, fyi @alexdima

jrieken committed:
Only layout breadcrumbs when needed, workaround for https://github.com/microsoft/vscode/issues/139877

joaomoreno committed:
fixes #139519

benoitf committed:
Rebase against the upstream cc64490e527b6b64f2e16c723b7c31026d63ecf6
vscode-upstream-sha1: cc64490e527b6b64f2e16c723b7c31026d63ecf6

benoitf commented:
@azatsarynnyy do you know why Roman was not added in the list of reviewers automatically ?

benoitf committed:
chore: use full path when importing modules

azatsarynnyy commented:
> @azatsarynnyy do you know why Roman was not added in the list of reviewers automatically ?

Most likely, it's because he doesn't have access to the repository.

benoitf commented:
@azatsarynnyy ok I sent a new invite

benoitf committed:
chore: handle package rename
src/vs/server is now src/vs/server/node

also change alpine build to x64 build by default and listen on 0.0.0.0 by default in dev mode
